### PR TITLE
MiningState Statistics

### PR DESCRIPTION
**Pull request by fosskers**
This PR helps us monitor Mining Coordination activity.

### CODE REVIEW

File: src/Chainweb/Chainweb/MinerResources.hs [hunk 1]
```diff
@@ -75,21 +82,38 @@ withMiningCoordination logger enabled cutDb inner
...
+            pure ms
+        count <- readIORef c
+        atomicWriteIORef c 0
+        logFunction logger Info . JsonLog $ MiningStats (M.size ms) count (avgTxs m)
+
+    f :: Time Micros -> T3 a PrevTime b -> Bool
+    f ago (T3 _ (PrevTime p) _) = p > BlockCreationTime ago
+
+    avgTxs :: MiningState -> Int
+    avgTxs (MiningState ms) = summed `div` M.size ms
```
Comment (fosskers) [inline src/Chainweb/Chainweb/MinerResources.hs]: `div` should be close enough.

File: src/Chainweb/Chainweb/MinerResources.hs [hunk 2]
```diff
@@ -75,21 +82,38 @@ withMiningCoordination logger enabled cutDb inner
...
-            , _coordLimit = 10000 }
+            , _coordLimit = 2500
+            , _coord503s = c }
   where
-    prune :: TVar MiningState -> IO ()
-    prune t = runForever (logFunction logger) "Chainweb.Chainweb.MinerResources.prune" $ do
-        let !d = 600000000  -- 10 minutes
+    prune :: TVar MiningState -> IORef Int -> IO ()
+    prune t c = runForever (logFunction logger) "Chainweb.Chainweb.MinerResources.prune" $ do
+        let !d = 300000000  -- 5 minutes
```
Comment (fosskers) [inline src/Chainweb/Chainweb/MinerResources.hs]: Halved.

File: src/Chainweb/Chainweb/MinerResources.hs [hunk 3]
```diff
@@ -75,21 +82,38 @@ withMiningCoordination logger enabled cutDb inner
...
     | otherwise = do
         t <- newTVarIO mempty
-        fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
+        c <- newIORef 0
+        fmap snd . concurrently (prune t c) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
             , _coordState = t
-            , _coordLimit = 10000 }
+            , _coordLimit = 2500
```
Comment (fosskers) [inline src/Chainweb/Chainweb/MinerResources.hs]: Quartered.

File: src/Chainweb/Chainweb/MinerResources.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/Chainweb/MinerResources.hs]: Would it be better to return the size from before prune? We may be interested in the `max` value.

Comment (fosskers) [reply src/Chainweb/Chainweb/MinerResources.hs]: It is already the size from before prune :)

**Pull request merged**